src/session/client-certificate.c

```diff
@@ -248,16 +248,35 @@ sssd_map_certificate (const char *certificate, char** username)
       goto out;
     }
 
+  /* sssd 2.6.1 introduces certificate validation against the configured CA. If present, use that API */
   r = sd_bus_call_method (bus,
                           "org.freedesktop.sssd.infopipe",
                           "/org/freedesktop/sssd/infopipe/Users",
                           "org.freedesktop.sssd.infopipe.Users",
-                          "FindByCertificate",
+                          "FindByValidCertificate",
                           &err,
                           &user_obj_msg,
                           "s",
                           certificate);
 
+  /* fallback for sssd < 2.6.1; this is only secure for matching the full certificate, not mapping rules */
```

Comment:
I don't like how we fall back to this automatically if we're on older sssd and meanwhile the documentation has been updated to say "this works now!" with only a version warning at the bottom, after the instructions, that probably nobody will get around to reading.

I feel like we should either find a way to match only complete known certificates on older sssd (ie: disable ones that are merely signed by a CA), or not update the docs until this is widely supported and secure.

Another option might be a config flag to enable falling back to this API, but of course, that would break existing users.

Let's chat about how we manage this.

Reply:
I updated the release note in this PR to drop the "certmap works now" paragraph. Indeed, let's only announce this once sssd 2.6.1 is available more broadly. The patches to the guide only became more strict -- of course I'm happy to adjust the wording in the warning, but it is still true that you can safely ignore the warning completely -- just as before, the guide *still* only documents how to set up complete certificate matching. For that, the only things that changed is (1) that you *must* set the CA in sssd now, and (2) it will now start to check CRL. Neither of this is "this works now", it just makes the whole thing more restrictive.

Due to that, I find that the automatic fallback is acceptable for now. I am not really worried about that at all, I am worried that using the validation API by default now is too much of a regression on upgrades. It's acceptable for Fedora and even RHEL 9, but not really for RHEL 8.6. I would like to refine that, but I need https://github.com/SSSD/sssd/issues/5911 fixed for that first -- then I can send a follow-up to fall back to the unvalidated API also on "sssd does not have a CA".